src/utils/include/utils/hash/Cantor.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef UTILS_HASH_CANTOR
+#define UTILS_HASH_CANTOR
+
+#include <utility>
+
+namespace Utils {
+namespace Hash {
+
+struct CantorPairing {
+  std::size_t cantor_pairing(int k1, int k2) const {
+    return static_cast<std::size_t>(0.5 * (k1 + k2) * (k1 + k2 + 1) + k2);
```

Comment:
This should be `((k1 + k2) * (k1 + k2 + 1)) / 2` to avoid the promotion to double, then also the cast can be removed.